srcStatic/ModuleLoader.cpp

```diff
@@ -26,6 +29,41 @@ void ModuleLoader::RegisterBuiltInModules()
 	}
 }
 
+void ModuleLoader::RegisterConsoleModules()
+{
+	if (consoleModuleNames.empty()) {
+		return; // No console modules to load
+	}
+
+	// Temporary check. This will eventually become an error.
+	if (consoleModuleNames.size() == 1 && consoleModuleNames[0].empty()) {
+		return; // No console modules to load
+	}
+
+	if (std::any_of(consoleModuleNames.begin(), consoleModuleNames.end(), [](const std::string& consoleModuleName) { return consoleModuleName.empty(); })) {
+		PostError("All console module names must be non-empty.");
+		return;
+	}
+
+	std::vector<std::string> moduleDirectories;
+	moduleDirectories.reserve(modules.size());
+
+	for (const auto& moduleName : consoleModuleNames)
```

Comment:
To be consistent with the old code, the `reserve` should perhaps be using `consoleModuleNames.size()`.

There may be cause to update this to include directories for other module types so this isn't necessarily bad. Though I also had some thoughts to allowing multiple `ResourceSearchPath` objects, which could be loaded and unloaded independently. That would allow for separate instances to manage search paths for console modules versus other sources. Hence we may want to keep this tied to the number of console modules anyway.